### DISCUSSION TITLE
Script Checks: Multiple Alert Actions

### DISCUSSION THREAD

**Discussion by hrmtc** [amidaware/tacticalrmm#2077] [Q&A] [answered]
My goal is to use Script Checks to verify if certain applications are installed, and if they are not, execute the corresponding script to install them.
However, how can I apply multiple Alert Templates to my agents? It only allows me to apply one at a time.

![image](https://github.com/user-attachments/assets/5b9c36f7-889e-4b02-8abf-5adb1b15f9fc)

Thanks!

**Comment by dinger1986** [comment]
Add more than one check and corresponding tasks to the template and assign it

**Reply by hrmtc** [reply]
Sorry, but I don't fully understand.  

Yes, I have multiple checks configured. For example, one to check if RustDesk is installed and another for BGInfo.

![image](https://github.com/user-attachments/assets/31bcc489-7e82-4371-97f4-f0852e5e91cf)

And in Alerts Manager, I have configured two Alert Actions so that when the Script Check output is an Error, it executes the corresponding script.

![image](https://github.com/user-attachments/assets/2c505924-c914-4f28-b1de-9fd777f86fe8)

My problem is that in Automation Manager, I can only choose one of the two templates.

Am I doing something wrong?

Thanks for your time.

![image](https://github.com/user-attachments/assets/2f29ea12-f906-47b1-af06-1d9f893c46c9)

**Reply by wh1te909** [reply, marked as answer]
Alerts manager is intended to only create alerts/notifications, not really run scripts on agents. While that is technically possible it's not the recommended way to do things, as the script that runs should only be a script to like send an email or a webhook or something.

What you want to do instead, is go to the "Tasks" tab in automation manager, create a task, choose your "failure" script and then for the scheduling options, choose "on check failure" and link it to the check. That way when the check fails, it will trigger that script to run.

**Reply by hrmtc** [reply]
Oh, now I understand! 

Thank you very much for your explanation! Now I can set up my actions with the corresponding scripts.

Have a great weekend!
